test/utils.js

```diff
@@ -878,3 +880,47 @@ describe('utils.addTimeoutToPromise()', () => {
         }
     });
 });
+
+describe('utils.infiniteScroll()', () => {
+    it('exits after no more to scroll', () => {
+        (async () => {
+            const browser = await puppeteer.launch({
+                headless: true,
+            });
+            const page = await browser.newPage();
+            const contentHTML = '<div>nothing</div>';
+            await page.setContent(contentHTML);
+            await utils.infiniteScroll(page);
+            await browser.close();
+        })();
+    });
+
+    it('exits after reaches the bottom', () => {
+        (async () => {
+            const browser = await puppeteer.launch({
+                headless: true,
+            });
+            const page = await browser.newPage();
+            await page.goto('https://twitter.com/search?src=typd&q=%23fingervein&lang=sv', {
+                waitUntil: 'networkidle2',
+            });
+            await utils.infiniteScroll(page);
+            await browser.close();
+        })();
+    });
+
+    it('times out if limit is set', () => {
+        (async () => {
+            const browser = await puppeteer.launch({
+                headless: true,
+            });
+            const page = await browser.newPage();
+            await page.goto('https://medium.com/search?q=biometrics', {
```

Comment:
The tests shouldn't depend on external websites, otherwise the tests will be flakey. I know this might be tricky sometimes, but maybe you can find some way? If not, please add a comment at least

Reply:
@vdrmota We now have a deployment pipeline in place that runs all tests before deploy. If the test is flaky, it will kill the pipeline. So it would be great if we could do this some other way. Perhaps add a new folder to SDK called `integration_tests` and a script `npm run test:integration` that will run those?